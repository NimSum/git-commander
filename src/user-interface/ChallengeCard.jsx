import React from 'react';

export function ChallengeCard (props) {
  return (
    <article className="challenge-card">
      <div>
        <p>{props.challenge.statement}</p>
      </div>
      <div>
        <p>{props.challenge.detail}</p>
      </div>
    </article>
  )
}