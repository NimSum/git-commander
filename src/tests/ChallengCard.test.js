import React from 'react';
import ChallengeCard from '../user-interface/ChallengeCard';
import Adapter from 'enzyme-adapter-react-16';
import { shallow , configure } from 'enzyme';

configure({adapter: new Adapter()});

describe('Challenge card' , () => {

  const mockQuestion =  {
    "statement" : "Our Git spaceship needs to switch to a different branch to steer the ship!",
    "detail" : "Switch to the branch called \"steer-ship\" to avoid the obstacle",
    "answer" : "git checkout steer-ship", 
    "difficulty" : "1",
    "moreInfo" : "/////A link to where they could read more about this command",
    "id": "1"
  };

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      < ChallengeCard 
        challenge={mockQuestion} />
    );
  });

  it('Should render a new challenge', () => {
    expect(wrapper).toMatchSnapshot();
  })

})