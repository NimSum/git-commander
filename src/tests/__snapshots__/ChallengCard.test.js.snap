// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Challenge card Should render a new challenge 1`] = `
<article
  className="challenge-card"
>
  <div>
    <p>
      Our Git spaceship needs to switch to a different branch to steer the ship!
    </p>
  </div>
  <div>
    <p>
      Switch to the branch called "steer-ship" to avoid the obstacle
    </p>
  </div>
</article>
`;
